meteor/client/ui/Shelf/DashboardPanel.tsx

```diff
@@ -568,79 +570,53 @@ export function getUnfinishedPieceInstancesReactive(playlist: RundownPlaylist) {
 	let prospectivePieces: PieceInstance[] = []
 	const now = getCurrentTime()
 	if (playlist.activationId && playlist.currentPartInstanceId) {
-		prospectivePieces = PieceInstances.find({
-			startedPlayback: {
-				$exists: true,
-			},
-			playlistActivationId: playlist.activationId,
-			$and: [
-				{
-					$or: [
-						{
-							stoppedPlayback: {
-								$eq: 0,
-							},
-						},
-						{
-							stoppedPlayback: {
-								$exists: false,
-							},
-						},
-					],
-				},
-				{
-					$or: [
-						{
-							adLibSourceId: {
-								$exists: true,
-							},
-						},
-						{
-							'piece.tags': {
-								$exists: true,
-							},
-						},
-					],
-				},
-				{
-					$or: [
-						{
-							userDuration: {
-								$exists: false,
-							},
-						},
-						{
-							'userDuration.end': {
-								$exists: false,
-							},
-						},
-					],
-				},
-			],
-		}).fetch()
+		const partInstance = PartInstances.findOne(playlist.currentPartInstanceId)
+
+		if (partInstance) {
+			const rundown = Rundowns.findOne(partInstance.rundownId)
+			if (rundown) {
+				const showStyle = rundown.getShowStyleBase()
+
+				prospectivePieces = PieceInstances.find({
```

Comment:
This seems like a very wide net, is the performance good with this change?

Reply:
I honestly don't know how to check, any tips? I get pretty bad stuttering and high cpu spikes locally before this change too.

Apart from omitting fields, I don't think we can safely filter out any pieces from this without risking differences in the operation

Reply:
Yeah, well, that's what I've always warned about when the Infinites changes were implemented: some major heavy lifting needs to be done by the UI anytime anything changes. I have not figured out a way to make it less heavy, I think, unfortunately, that's just a part of this Infinites architecture.

Okay, I started to think about doing like a ReactiveVar or something with a React Context to at least share this data across components, but I think there's actually a simpler (at least implementation-wise) way: wrap the entire `getUnfinishedPieceInstancesReactive` in a `memoizedIsolatedAutorun`. The only thing it depends on is the playlistId, so that can be easily piped in as a parameter and then at least we'll only get one of these calculations for the entire view.

Reply:
I think I shall need some help with configuring this properly. I tried doing:
```
return memoizedIsolatedAutorun(
 	(playlistActivationId, currentPartInstanceId, showStyleBase) => {
.....
},
	'getUnfinishedPieceInstancesReactive',
	playlist.activationId,
	playlist.currentPartInstanceId,
	showStyleBase
)

but that is not invalidating correctly when I run an adlib. Any suggestions?